std/range/package.d

```diff
@@ -10195,13 +10195,18 @@ in
             auto result = adds(start, range.length, overflow);
         else static if (isSigned!Enumerator)
         {
-            Largest!(Enumerator, Signed!LengthType) signedLength;
+            alias signed_t = Largest!(Enumerator, Signed!LengthType);
+            signed_t signedLength;
+            //Can length fit in the signed type
+            overflow = range.length > signed_t.max;
+            /*
+            //This isn't nogc(but should be...)
             try signedLength = to!(typeof(signedLength))(range.length);
```

Comment:
This line also assigns `signedLength` to `range.length`, however this is omitted in the proposed change. Is this not going to affect the code?

Reply:
I don't think it effects the actual behaviour of anything apart from the check in the static if. 

I'll fix it - there's no test case for this so I'll try and come up with one.